sp/src/public/vscript/ivscript.h

```diff
@@ -163,20 +163,6 @@ class IScriptManager : public IAppSystem
 // 
 //-----------------------------------------------------------------------------
 
-#ifdef MAPBASE_VSCRIPT
-template <typename T> T *HScriptToClass( HSCRIPT hObj )
-{
-	return (hObj) ? (T*)g_pScriptVM->GetInstanceValue( hObj, GetScriptDesc( (T*)NULL ) ) : NULL;
```

Comment:
`IScriptVM` is not yet defined here. MSVC probably accepts this as long as the template is not expanded yet.
Moving it below the definition of `IScriptVM` seemed to cause no harm and fixes the gcc build.